src/Val.h

```diff
@@ -1176,13 +1176,15 @@ class RecordVal final : public Val, public notifier::detail::Modifiable {
 	/**
 	 * Appends a value to the record's fields.  The caller is responsible
 	 * for ensuring that fields are appended in the correct order and
-	 * with the correct type.
+	 * with the correct type.  The type needs to be passed in because
+	 * it's unsafe to take it from v when the field's type is "any" while
+	 * v is a concrete type.
 	 * @param v  The value to append.
 	 */
-	void AppendField(ValPtr v)
+	void AppendField(ValPtr v, const TypePtr& t)
```

Comment:
This is a general question, but I just noticed on this method. Is there any concern around deprecation with changes like this? `RecordVal` is not in a `detail` namespace, so if public methods are changing we need to follow the deprecation process.

Reply:
Good question.  That method was introduced in Nov 2020, and it really is only meaningful for internal use - calling it requires tricky coordination regarding the record's internal state.  It should be protected, and I don't know why I didn't make it so.  So my own druthers would be to pretend it was never non-protected and change it now.  But if that's untenable, then we can add an overload and leave the original (though the original is buggy for "any" types, hence the change to it here).

Reply:
Nope, making it protected would be fine with me.

Reply:
Roger, added that to an upcoming push.